### PR TITLE
Add scopes, scope modification, and scope validation

### PR DESCRIPTION
**Pull request by jsachs**
#### What's this PR do?
Allows users to create OAuth-style scope arrays for permissions embedded in JWT. Sets up middleware for the microservice to validate admin functionality, which serves as an example of how to validate scopes in other services.

#### Related JIRA tickets:
SER-10, SER-43

#### How should this be manually tested?
Create a user with `admin` scope. Make sure this user can update user scopes, as this is an admin-protected function. The workflow is demonstrated in the test code.

### CODE REVIEW

File: server/routes/user.route.js [hunk 1]
```diff
@@ -23,6 +25,13 @@ router.route('/:userId')
     /** DELETE /api/users/:userId - Delete user */
     .delete(userCtrl.remove);
 
+router.route('/scopes/:userId')
+    /** PUT /api/users/scopes/:userId - Update user scopes */
```
Comment (Perry5) [inline server/routes/user.route.js]: tiny thing, but this should be `/api/user/scopes/:userId` in the comment

Comment (jsachs) [reply server/routes/user.route.js]: Good catch!

**Comment by jsachs** [comment]
@sirctseb I don't think this is true. Look at `user.integration.spec.js` line 236. The logged in user is the admin user, but it is using the userId set by creating the non-admin user. This works because the `req.user` prop is set by the `param()` middleware, not the `jwt` middleware. If you have a test case that proves otherwise let me know, but I think this is sound.

**Comment by sirctseb** [comment]
Chain this comparison of the intended scopes value to data from an explicit get request for the user
```javascript
                .then(() => {
                    request(app)
                    .get(`${baseURL}/user/${userId}`)
                    .then((userRes) => {
                        expect(userRes.body.scopes).to.deep.equal(['newScope']);
                        return;
                    })
                    .catch(done);
                })
```

**Comment by jsachs** [comment]
@sirctseb good catch, fix is posted. I've also included your change to the unit test.

**Comment by sirctseb** [comment]
Looks good. We might also leave the test that checked the response body from the PUT request itself.

**Pull request merged**